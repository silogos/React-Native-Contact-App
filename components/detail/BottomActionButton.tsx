import React from "react";
import { Pressable, PressableProps, StyleSheet, View } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import { ThemedIcon } from "../ThemedIcon";
import { ThemedText } from "../ThemedText";

interface BottomActionButtonType extends PressableProps {
  iconName: keyof typeof Ionicons.glyphMap;
  title: String;
}

const BottomActionButton = React.forwardRef<View, BottomActionButtonType>(
  ({ iconName, title, ...props }, ref) => {
    return (
      <Pressable {...props} style={styles.container} ref={ref}>
        <ThemedIcon size={20} name={iconName} />
        <ThemedText type="default" style={styles.title}>
          {title}
        </ThemedText>
      </Pressable>
    );
  }
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    gap: 5,
  },
  title: {
    fontSize: 10,
  },
});

export default BottomActionButton;
