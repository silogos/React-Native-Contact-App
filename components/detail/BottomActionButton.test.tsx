import React from "react";
import { render } from "@testing-library/react-native";

import BottomActionButton from "./BottomActionButton";

describe("Detail", () => {
  it("renders correctly", () => {
    const elm = render(
      <BottomActionButton iconName="accessibility" title={"Delete"} />
    ).toJSON();

    expect(elm).toMatchSnapshot();
  });

  it("Render with title", () => {
    const title = "Delete";
    const { getByText } = render(
      <BottomActionButton iconName="accessibility" title={title} />
    );

    expect(getByText(title)).toBeTruthy();
  });
});
