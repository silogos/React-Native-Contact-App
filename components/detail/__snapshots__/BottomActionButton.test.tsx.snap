// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "gap": 5,
      "justifyContent": "center",
    }
  }
>
  <Text />
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        {
          "fontSize": 10,
        },
      ]
    }
  >
    Delete
  </Text>
</View>
`;
